components/debugbar/logs.tsx

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo} from 'react';
+import {useSelector} from 'react-redux';
+import cn from 'classnames';
+import {FixedSizeList as List} from 'react-window';
+
+import {getLogs} from 'mattermost-redux/selectors/entities/debugbar';
+
+import {DebugBarLog} from '@mattermost/types/debugbar';
+
+import Code from './code';
+import Time from './time';
+
+type Props = {
+    filter: string;
+    height: number;
+}
+
+type RowProps = {
+    data: DebugBarLog[];
+    index: number;
+    style: any;
+}
+
+function Row({data, index, style}: RowProps) {
+    return (
+        <div
+            key={data[index].time + '_' + data[index].message}
+            className='DebugBarTable__row'
+            style={style}
+        >
+            <div className={cn('time', {error: data[index].level === 'error'})}><Time time={data[index].time}/></div>
+            <div className='logMessage'>{data[index].message}</div>
+            <div
+                className='json'
+                title={JSON.stringify(data[index].fields)}
+            >
+                <Code
+                    code={JSON.stringify(data[index].fields)}
```

Comment:
Nit: There's enough server logs that it might be worth stringifying the data only once and storing it in a constant. That said, things feel reasonably snappy on my local at the moment.

Reply:
I think that we can use a client side indexing library and subscribe to the store to generate the indexes, that way we also have way better search support